CHANGES.rst

```diff
@@ -7,6 +7,12 @@ Unreleased
 
 -   Update Click dependency to >= 8.0.
 
+-   Fixed the issue where typing requires ``template global``
+    decorators to accept functions with no arguments. This meant
+    user defined functions which require to take in arguments
+    fail the lint tests (eg - ``mypy``).
+    Fixes ``#4098``
```

Comment:
Please follow the style used in the rest of the changelog for linking to issues. Also, can remove the section describing that the incorrect type caused type tests to fail, only the first sentence is needed. Also, `template global` is a phrase, not code, so it shouldn't be in code formatting.